src/nvim/eval.c

```diff
@@ -10767,6 +10767,25 @@ static void f_getwininfo(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   }
 }
 
+/// "waituntil(condition, timeout)" function
```

Comment:
maybe `timeout` should go first, since the condition/lambda/etc is often somewhat long. OTOH this forces `timeout` to always be specified, but it's probably good to force that decision anyways: waiting "forever" is probably not common.

Reply:
"forever" is useful if you don't need to impose an extra limit because you are sure the event will happen soon anyway (and if you're wrong the user can still Ctrl-C). But we can use 0 to indicate that.

Reply:
> But we can use 0 to indicate that.

yes, although I would suggest `-1` since that is consistent with `timer_start(...,{'repeat':-1})` . 

And `0` means "check without waiting" in `jobwait()`.

Reply:
indeed, `jobwait([...], -1)` seems to also work.